src/components/NewCommentForm.tsx

```
@@ -1,8 +1,82 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { Comment } from '../types/Comment';
+import { Post } from '../types/Post';
+import PropTypes from 'prop-types';
+
+type Props = {
+  onSubmit: (comment: Omit<Comment, 'id'>) => Promise<void>;
+  selectedPost?: Post;
+};
+
+type FormState = {
+  name: string;
+  email: string;
+  body: string;
+};
+
+export const NewCommentForm: React.FC<Props> = ({ onSubmit, selectedPost }) => {
+  const [form, setForm] = useState<FormState>({
+    name: '',
+    email: '',
+    body: '',
+  });
+
+  const [errors, setErrors] = useState<FormState>({
+    name: '',
+    email: '',
+    body: '',
+  });
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  function handleChange<K extends keyof FormState>(key: K, value: string) {
+    setForm(prev => ({ ...prev, [key]: value }));
+    setErrors(prev => ({ ...prev, [key]: '' }));
+  }
+
+  function clearForm() {
+    setForm({ name: '', email: '', body: '' });
+    setErrors({ name: '', email: '', body: '' });
+    setIsLoading(false);
+  }
+
+  function resetBody() {
+    setForm(prev => ({ ...prev, body: '' }));
+    setErrors(prev => ({ ...prev, body: '' }));
+  }
+
+  async function handleSubmit(e: React.FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+
+    if (!selectedPost?.id) {
+      return;
+    }
+
+    const newErrors: FormState = {
+      name: form.name.trim() ? '' : 'Name is required',
+      email: form.email.trim() ? '' : 'Email is required',
+      body: form.body.trim() ? '' : 'Comment is required',
+    };
+
+    setErrors(newErrors);
+
+    if (Object.values(newErrors).some(Boolean)) {
+      return;
+    }
+
+    try {
+      setIsLoading(true);
+      await onSubmit({ ...form, postId: selectedPost.id });
+      resetBody();
+    } catch {
```

The catch block is empty which swallows any errors from the onSubmit API call. Consider at least logging the error and/or setting an error state so the app can show a user-facing message or allow retry. For example, capture the error and call a prop like `onError` or set local state `setSubmitError(error.message)` so the UI can display `CommentsError` or similar. This aligns with the checklist suggestion to use try/catch around async calls for robust error handling.